prosopopee/prosopopee.py

```diff
@@ -229,10 +229,155 @@ def main():
+
+    if not os.path.exists(os.path.join("build", gallery_path)):
+            os.makedirs(os.path.join("build", gallery_path))
+
+    if gallery_settings.get("public", True) or dirs:
+        error(gallery_settings.get("title"), "Your gallery describe in %s need to have a "
+                                             "title" % (os.path.join(gallery, "settings.yaml")))
+        error(gallery_settings.get("cover"), "You should specify a path to a cover picture "
+                                             "in %s" % (os.path.join(gallery, "settings.yaml")))
+
+        if isinstance(gallery_settings["cover"], dict):
+            cover_image_path = os.path.join(gallery_path, gallery_settings["cover"]["name"])
+            cover_image_url = os.path.join(gallery, gallery_settings["cover"]["name"])
+            cover_image_type = gallery_settings["cover"]["type"]
+        else:
+            cover_image_path = os.path.join(gallery_path, gallery_settings["cover"])
+            cover_image_url = os.path.join(gallery, gallery_settings["cover"])
+            cover_image_type = "image"
+
+        error(os.path.exists(cover_image_path), "File for %s cover image doesn't exist at "
+                                                "%s" % (gallery, cover_image_path))
+
+        gallery_cover = {
+            "title": gallery_settings["title"],
+            "link": gallery,
+            "sub_title": gallery_settings.get("sub_title", ""),
+            "date": gallery_settings.get("date", ""),
+            "tags": gallery_settings.get("tags", ""),
+            "cover_type": cover_image_type,
+            "cover": cover_image_url,
+        }
+
+        if dirs:
+            subgallery_theme = gallery_settings.get("theme")
+            if subgallery_theme:
+                theme_path = os.path.exists(os.path.join(os.path.split(os.path.realpath(__file__))[0], "themes", subgallery_theme))
+                available_themes = subgallery_theme, "', '".join(os.listdir(os.path.join(os.path.split(os.path.realpath(__file__))[0],
+                                                                              "themes")))
+
+                error(theme_path, "'%s' is not an existing theme, available themes are '%s'" % available_themes)
+
+                templates_dir = [
+                    os.path.realpath(os.path.join(os.getcwd(), "templates")),
+                    os.path.join(os.path.split(os.path.realpath(__file__))[0], "themes", subgallery_theme, "templates")
+                ]
+
+                if subgallery_theme != "exposure":
+                    templates_dir.append(os.path.join(os.path.split(os.path.realpath(__file__))[0],
+                                                      "themes", "exposure", "templates"))
+
+                subgallery_templates = Environment(loader=FileSystemLoader(templates_dir))
+            else:
+                subgallery_theme = settings.get("theme", "exposure")
```

Comment:
This line is already called line 287, both could be merged/simplified (including the condition).